### DISCUSSION TITLE
Welcome to javapoet Discussions!

### DISCUSSION THREAD

**Discussion by JakeWharton** [square/javapoet#865] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by yaccoff** [comment]
Hi there, Is there a way to append a comment to a item declaration: I am using JavaDoc comments currently and they take up 3 lines:
    /**
     * BUFFER OVERRUN INDICT
     */
    public final List<Integer> AABOR;
I want:
    public final List<Integer> AABOR; /** BUFFER OVERRUN INDICT */

**Reply by JakeWharton** [reply]
There is not, no.

**Reply by yaccoff** [reply]
Okay thx for replying. Andy

**Reply by yaccoff** [reply]
On second thoughts, any reason I cant add it?

**Reply by JakeWharton** [reply]
We do not want such a feature because it has a lot of pitfalls. The most critical of which is what happens when the comment contains a newline?

**Reply by yaccoff** [reply]
Ah just reiwing the other post, about off loading.

**Reply by yaccoff** [reply]
I dont understand why it has pitfalls, but your more involved than I to understand that. My problem is that I am translating Lang A to Lang B and there are 10 of thousands of variables, all with 3 lines of Javadoc.

**Comment by ronshapiro** [comment]
FieldSpec.addInitializer("; // BUFFER OVERRUN INDICT") is a hack that would
work - it would append a semicolon at the end of the comment if you're ok
with that.

But generated code shouldn't really care about extra lines, so none of this
is likely worth it.

**Reply by yaccoff** [reply]
Can I just understand your reply: Given this code, 

FieldSpec fs = FieldSpec.builder(listParameterizedTypeName, name)
		.addModifiers(Modifier.PUBLIC, Modifier.FINAL)
		.addJavadoc(strJavadoc)
		.build();

it would change to this?

FieldSpec fs = FieldSpec.builder(listParameterizedTypeName, name)
		.addModifiers(Modifier.PUBLIC, Modifier.FINAL)
.addInitializer("; // BUFFER OVERRUN INDICT")
		.addJavadoc(strJavadoc)
		.build();

**Reply by yaccoff** [reply]
Minus the addJavadoc line -;)

**Comment by ronshapiro** [comment]
Replace addJavadoc and use addInitializer("; $L", strJavadoc)

**Reply by yaccoff** [reply]
Got you, thx Ron

**Comment by yaccoff** [comment]
Having implemented the suggestion; by adding:

initialise("ArrayList\<Integer\>(); // $L", strJavaDoc) ......the ArrayList isnt imported into the class definition, Any way to do that?

**Comment by ronshapiro** [comment]
I think you'd do well to read the docs

בתאריך יום ו׳, 25 בפבר׳ 2022, 12:07, מאת yaccoff ***@***.***
